### PR TITLE
readme update for NNFrames Scala examples

### PR DESCRIPTION
**Pull request by hhbyyh**
also update the examples for consistency and correct some errors.

### CODE REVIEW

**Comment by hhbyyh** [comment]
Hi @dding3 can you help review this? Thanks.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/ImageFinetune.scala [hunk 1]
```diff
@@ -25,82 +25,53 @@ import com.intel.analytics.zoo.pipeline.api.keras.models.Model
...
       val sc = NNContext.initNNContext(conf)
 
-      val model = getTransferLearningModel(param.modelPath)
-
-      val (trainDf, valDf) = getImageData(param.dataPath, sc)
+      val createLabel = udf { row: Row =>
+        if (new Path(row.getString(0)).getName.contains("cat")) 1.0 else 2.0
+      }
+      val imagesDF = NNImageReader.readImages(param.imagePath, sc,
+          resizeH = 300, resizeW = 300, imageCodec = 1)
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/ImageFinetune.scala]: why do we need resize here? Looks like in the following transformer there is a resize too?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md [hunk 2]
```diff
@@ -1,51 +1,59 @@
...
 
-1. Download the pre-trained model
+## Run the example
 
-You can download the pre-trained BigDL model [here](https://github.com/intel-analytics/analytics-zoo/tree/legacy/models).
+1. Download the pre-trained model
 
-In this example, we are going to use the Inception-V1 model. Please feel free to use other models.
+You can download the pre-trained BigDL model
+[here](https://github.com/intel-analytics/analytics-zoo/tree/legacy/models), and put it
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md]: The link is from legacy branch, change it to https://analytics-zoo.github.io/master/#ProgrammingGuide/image-classification/#download-link?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/imageInference/ImageInferenceExample.scala [hunk 3]
```diff
@@ -35,20 +39,21 @@ object ImageInferenceExample {
       val sc = NNContext.initNNContext("ImageInference")
 
       val getImageName = udf { row: Row => row.getString(0)}
-      val imageDF = NNImageReader.readImages(params.folder, sc)
+      val imageDF = NNImageReader.readImages(params.imagePath, sc,
+          resizeH = 300, resizeW = 300, imageCodec = 1)
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/imageInference/ImageInferenceExample.scala]: why need resize here?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/imageTransferLearning/ImageTransferLearning.scala [hunk 4]
```diff
@@ -39,16 +45,17 @@ object ImageTransferLearning {
       val sc = NNContext.initNNContext()
 
       val createLabel = udf { row: Row =>
-        if (row.getString(0).contains("demo/cats")) 1.0 else 2.0
+        if (new Path(row.getString(0)).getName.contains("cat")) 1.0 else 2.0
       }
-      val imagesDF: DataFrame = NNImageReader.readImages(params.folder + "/*/*", sc)
+      val imagesDF: DataFrame = NNImageReader.readImages(params.imagePath, sc,
+          resizeH = 300, resizeW = 300, imageCodec = 1)
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/imageTransferLearning/ImageTransferLearning.scala]: why resize?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/ImageFinetune.scala (hunk 1 above)
Comment (hhbyyh) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/ImageFinetune.scala]: make the image cache smaller.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md (hunk 2 above)
Comment (hhbyyh) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md]: Sure. Thanks

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/ImageFinetune.scala (hunk 1 above)
Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/ImageFinetune.scala]: Can we resize the image to 256*256 so we don't need the ImageResize?

Comment (hhbyyh) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/ImageFinetune.scala]: OK Updated.

**Comment by hhbyyh** [comment]
@dding3 Thanks for the review. Let me know if there's further comments.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md [hunk 5]
```diff
@@ -1,77 +1,88 @@
...
 
-1. Download the pre-trained model
+## Run the example
 
-You can download the pre-trained BigDL model [here](https://github.com/intel-analytics/analytics-zoo/tree/legacy/models).
+1. Download the pre-trained model
 
-In this example, we are going to use the Inception-V1 model. Please feel free to use other models.
+You can download the pre-trained model from
+[Analytics Zoo](https://s3-ap-southeast-1.amazonaws.com/bigdl-models/imageclassification/imagenet/bigdl_inception-v1_imagenet_0.4.0.model), and put it in `/tmp/zoo` or other path.
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md]: Can we put the zoo model download link(https://analytics-zoo.github.io/master/#ProgrammingGuide/image-classification/#download-link) here? Users may want to try other model besides inception.

**Comment by dding3** [comment]
Thanks for your update. It looks good to me except the one minor comment about the download zoo link.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md (hunk 5 above)
Comment (hhbyyh) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/nnframes/finetune/README.md]: I can add it to the end of the notebook and encourage users to try other models. Thanks.

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1226/ passed. Merging to master. Thanks for the review

**Pull request merged**